bin/create_mbr_disk.ml

```diff
@@ -0,0 +1,70 @@
+open Cmdliner
+
+let create_mbr_disk destination partition_files =
+  let sector_size = Mbr.sizeof in
+  let num_partitions = List.length partition_files in
+  if num_partitions > 4 then
+    failwith
+      "Too many partition files. Limit number of files to 4 as MBR supports at \
+       most 4 partitions"
+  else Printf.printf "Total partitions to create: %d\n" num_partitions;
+
+  let partition_sizes =
+    List.map (fun file -> (Unix.stat file).Unix.st_size) partition_files
+  in
+
+  let partition_size = List.fold_left ( + ) 0 partition_sizes in
+  let mbr_size = Mbr.sizeof in
+  let total_size = partition_size + mbr_size in
+
+  Printf.printf "Total disk size: %d bytes\n" total_size;
+
+  let partitions =
+    List.mapi
+      (fun i size ->
+        Printf.printf "Creating partition: %d" (i + 1);
+        let start_sector = (i + 1) * sector_size in
+        let num_sectors = (size + sector_size - 1) / sector_size in
```

Comment:
@reynir Here I wish to ask, the files to be written to the disk, should they have a size that is an exact multiple of `sector_size` before it is validated and we error otherwise.

I agree too much slack at the end of the partition will be a waste.

Reply:
Yes, I would do it that way. The user can call `truncate` to add padding if they need it and it makes sense for the data in question.